include/ofxColour.h

```diff
@@ -0,0 +1,227 @@
+   - Default - kOfxImageEffectPropColourManagementStyleNone
+   - Valid Values - This must be one of
+     - ::kOfxImageEffectPropColourManagementNone - no colour management
+     - ::kOfxImageEffectPropColourManagementBasic - only basic colourspaces from ofxColourspaceList.h may be used
+     - ::kOfxImageEffectPropColourManagementCore - only core colourspaces from ofxColourspaceList.h may be used
+     - ::kOfxImageEffectPropColourManagementFull - any colourspace from ofxColourspaceList.h may be used
+     - ::kOfxImageEffectPropColourManagementOCIO - any OCIO config may be used (implies use of the OCIO library)
+
+Hosts should set this property if they will provide colourspace information 
+to plug-ins.
+Plug-ins should set this property if they can use host-provided colourspace 
+information.
+Collectively, the full, core and basic styles are referred to as native
+colour management. OCIO is used as the reference for the colour management
+API, but is not required to implement the native styles.
+
+The colourspace strings used in the native styles are from
+openfx-studio-config-v2.1.0_acces-v1.3_ocio-v2.3.ocio
+and stored for OFX purposes in ofxColourspaceList.h. Additionally, there is
+a scheme for cross-referencing between clips.
+
+The assumption is that OCIO > Full > Core > Basic, so the highest style
+supported by both host and plug-in will be chosen.
+*/
+#define kOfxImageEffectPropColourManagementStyle "OfxImageEffectPropColourManagementStyle"
+
+/* String used to indicate that no colour management is available. */
+#define kOfxImageEffectPropColourManagementNone "OfxImageEffectPropColourManagementNone"
+/* String used to indicate that basic colour management is available. */
+#define kOfxImageEffectPropColourManagementBasic "OfxImageEffectPropColourManagementBasic"
+/* String used to indicate that core colour management is available. */
+#define kOfxImageEffectPropColourManagementCore "OfxImageEffectPropColourManagementCore"
+/* String used to indicate that Full colour management is available. */
+#define kOfxImageEffectPropColourManagementFull "OfxImageEffectPropColourManagementFull"
+/* String used to indicate that OCIO colour management is available. */
+#define kOfxImageEffectPropColourManagementOCIO "OfxImageEffectPropColourManagementOCIO"
+
+/** @brief The path to the OCIO config used for this instance
+
+   - Type - string X 1
+   - Property Set - image effect instance (read only)
+   - Valid Values - Filesystem path to the config or URI starting ocio://
+
+A host must set this property on any effect instances where it has negotiated 
+OCIO colour management (kOfxImageEffectPropColourManagementOCIO).
+Use of URIs for built-in configs, such as ocio://default is permitted.
+
+When native colour management is in use, a host must set this
+property to point to the OCIO config used to define strings for the version of
+OFX it was built against. This will allow a plug-in which uses OCIO to work
+directly with native styles.
```

Comment:
I've updated this comment to include the filename of the config, but also explained that a host could use a different config, provided it included all the same definitions as the standard config. For example they might have a config which is a superset of the native config.

Reply:
Interesting, I had misunderstood how you were intending this to work for the two scenarios where either host is OCIO & plug-in is Native, or host is Native & plug-in is OCIO.

What I had assumed was that whichever half was Native would provide an ID string (not a path) which identifies a specific config from an OFX release. It would then be the responsibility of those implementing OCIO in OFX to have as part of their install a config that could respond to those color spaces. The most straight-forward would be to just have a copy of the OFX config(s). But another scenario would be that it has a config which is a superset of the OFX config which adds additional color spaces relevant to the app or the user. In fact, it could use the OCIO merge feature (under development) to merge the OFX config into a config provided by the end-user.

But if I understand what you're saying, it's the Native half that would provide a path to a config it wants the OCIO one to use. I don't like that approach for two reasons. First, it pushes the burden of having an OCIO config onto the Native half of the scenario, which may have minimal understanding of OCIO and would not want to have to supply a config file. Secondly, it limits the flexibility of the OCIO half of the scenario to use a superset config that has additional color spaces that are meaningful to it. You seem to be saying that the Native half would want to specify a config that could have additional color spaces, but that doesn't make sense to me. In other words, if it's in Native mode, there's no way for it to identify any additional color spaces other than what's in the header file.

Or is it that only the host may set this?  I agree that if both host & plug-in are in OCIO mode, it should be the host that determines the config, and it should be a path rather than an ID string. But if only one of the two is OCIO and the other is Native, it should be the Native one that calls out the color space set it is using as an ID string (currently only one set is available, but I'm assuming that a new set might be added with every OFX release).

Reply:
I discarded this idea and added separate properties to specify the native mode config. This means the OCIO property is back to just being for OCIO, removing this overloaded functionality and any requirements for non-OCIO hosts to include an OCIO config. Now if either plug-in or host is implemented using OCIO, but operating in native mode, it can get all the information it needs from these separate properties.

These new properties allow for versioning of the colourspace set (I'm using the OCIO term config for this) independently from the API version, using the OCIO config name as the id string.